File: Modules/SEDO/EMS/SIE.Web.EMS.Purchases/SparePartAcceptances/Commands/SaveAcceptAttachmentCommand.js
SIE.defineCommand('SIE.Web.EMS.Purchases.SparePartAcceptances.Commands.SaveAcceptAttachmentCommand', {
    extend: 'SIE.cmd.FormSave',
    meta: { text: "上传", group: "edit", iconCls: "icon-SaveEntity icon-blue" },
    canExecute: function (view) {
        return this.callParent(arguments);
    },
    execute: function (view, source) {
        var data = view.getCurrent().data;
        view.execute({
            withIds: true,
            data: data,
            success: function (rst) {
                var win = Ext.getCmp('UploadSpareAcceptCommand_Window');
                win.close();
                CRT.Event.fire("SIE.EMS.Purchases.SparePartAcceptances.SparePartAcceptance_refresh");
                Ext.MessageBox.alert("提示".t(), "上传成功".t());
            }
        });
    }
});

File: Modules/SEDO/EMS/SIE.Web.EMS.Purchases/SparePartAcceptances/Commands/SaveAcceptAttachmentCommand.cs
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.EMS.Purchases.SparePartAcceptances;
using SIE.Web.Command;
using System;

namespace SIE.Web.EMS.Purchases.SparePartAcceptances.Commands
{
    /// <summary>
    /// 保存命令
    /// </summary>
    [JsCommand("SIE.Web.EMS.Purchases.SparePartAcceptances.Commands.SaveAcceptAttachmentCommand")]
    public class SaveAcceptAttachmentCommand : FormSaveCommand
    {
        /// <summary>
        /// 执行命令
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            if (args == null)
            {
                throw new ValidationException("数据异常".L10N());
            }

            var data = args.Data.ToJsonObject<SparePartAcceptanceAttachment>();
            
            data.PersistenceStatus = PersistenceStatus.New;

            RF.Save(data);
            return true;
        }
    }
}
